src/backend/distributed/commands/table.c

```diff
@@ -165,31 +166,60 @@ PreprocessDropTableStmt(Node *node, const char *queryString,
 
 
 /*
- * PostprocessCreateTableStmt takes CreateStmt object as a parameter and errors
- * out if it creates a table with a foreign key that references to a citus local
- * table.
+ * PostprocessCreateTableStmt takes CreateStmt object as a parameter and
+ * processes foreign keys on relation via PostprocessCreateTableStmtForeignKeys
+ * function.
  *
  * This function also processes CREATE TABLE ... PARTITION OF statements via
  * PostprocessCreateTableStmtPartitionOf function.
  */
 void
 PostprocessCreateTableStmt(CreateStmt *createStatement, const char *queryString)
+{
+	PostprocessCreateTableStmtForeignKeys(createStatement);
+
+	if (createStatement->inhRelations != NIL && createStatement->partbound != NULL)
+	{
+		/* process CREATE TABLE ... PARTITION OF command */
+		PostprocessCreateTableStmtPartitionOf(createStatement, queryString);
+	}
+}
+
+
+/*
+ * PostprocessCreateTableStmtForeignKeys drops ands re-defines foreign keys
+ * defined by given CREATE TABLE command if command defined any foreign to
+ * reference or citus local tables.
+ */
+static void
+PostprocessCreateTableStmtForeignKeys(CreateStmt *createStatement)
 {
 	/*
 	 * Relation must exist and it is already locked as standard process utility
 	 * is already executed.
 	 */
 	bool missingOk = false;
 	Oid relationId = RangeVarGetRelid(createStatement->relation, NoLock, missingOk);
-	if (HasForeignKeyToCitusLocalTable(relationId))
-	{
-		ErrorOutForFKeyBetweenPostgresAndCitusLocalTable(relationId);
-	}
 
-	if (createStatement->inhRelations != NIL && createStatement->partbound != NULL)
+	int nonDistTableFKeysFlag = INCLUDE_REFERENCING_CONSTRAINTS |
```

Comment:
Needs comment on why we only need `INCLUDE_REFERENCING_CONSTRAINTS `

Reply:
actually we explain in a few lines below

https://github.com/citusdata/citus/pull/4480/files#diff-f9ddfd7904163c5cf85fdff5ab0444ea580676dcf59ab5bf4beddc6f65c57c45R212-R217

Reply:
I guess the last sentence should be moved before this line?

Reply:
makes sense, 5c02e26